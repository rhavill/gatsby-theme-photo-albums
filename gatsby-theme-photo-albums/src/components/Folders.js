import React from 'react'
import PropTypes from 'prop-types'
import Folder from './Folder'
import {mapIndexed} from '../util/ramda-utils'

const Folders = ({folders, icon}) => {
  return mapIndexed((folder, i) => 
    <Folder key={i} path={folder} icon={icon} />
  )(folders)
}

Folders.propTypes = {
  folders: PropTypes.arrayOf(
    PropTypes.string.isRequired, 
  ).isRequired,
  icon: PropTypes.shape({
    src: PropTypes.string.isRequired,
    srcSet: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired
}

export default Folders