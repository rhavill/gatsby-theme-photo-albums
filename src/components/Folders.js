import React from 'react'
import PropTypes from 'prop-types'
import addIndex from 'ramda/src/addIndex'
import compose from 'ramda/src/compose'
import map from 'ramda/src/map'
import Folder from './Folder'
import getChildPaths from '../util/files-folders'

const mapIndexed = addIndex(map)

const Folders = ({path, folders, icon}) => {
  return compose(
    mapIndexed((folder, i) => 
      <Folder key={i} path={folder} icon={icon} />
    ),
    getChildPaths(path)
  )(folders)
}

Folders.propTypes = {
  path: PropTypes.string.isRequired,
  folders: PropTypes.arrayOf(
    PropTypes.string.isRequired, 
  ).isRequired,
  icon: PropTypes.shape({
    src: PropTypes.string.isRequired,
    srcSet: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired
}

export default Folders