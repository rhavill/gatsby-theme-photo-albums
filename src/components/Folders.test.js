import React from "react"
import renderer from "react-test-renderer"
import {cleanup} from '@testing-library/react';
import Folders from "./Folders"
import folderData from '../test-data/source-filesystem-folder-data'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe("Folders", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Folders path='/level-one' data={folderData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})