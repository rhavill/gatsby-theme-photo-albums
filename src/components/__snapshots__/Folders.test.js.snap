// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folders renders correctly 1`] = `
Array [
  <article
    className="folder"
    role="folder"
  >
    <a
      href="/level-one/level-two"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": 250,
            "overflow": "hidden",
            "position": "relative",
            "width": 250,
          }
        }
      >
        <picture>
          <source
            srcSet="/static/hash/f17d3/folder.png 1x"
          />
          <img
            alt="Level Two"
            height={250}
            loading="lazy"
            onLoad={[Function]}
            src="/static/hash/f17d3/folder.png"
            srcSet="/static/hash/f17d3/folder.png 1x"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
            width={250}
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/hash/f17d3/folder.png 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/hash/f17d3/folder.png 1x\\" src=\\"/static/hash/f17d3/folder.png\\" alt=\\"Level Two\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <div
        className="folder-title"
      >
        Level Two
      </div>
    </a>
  </article>,
  <article
    className="folder"
    role="folder"
  >
    <a
      href="/level-one/another-level-two"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": 250,
            "overflow": "hidden",
            "position": "relative",
            "width": 250,
          }
        }
      >
        <picture>
          <source
            srcSet="/static/hash/f17d3/folder.png 1x"
          />
          <img
            alt="Another Level Two"
            height={250}
            loading="lazy"
            onLoad={[Function]}
            src="/static/hash/f17d3/folder.png"
            srcSet="/static/hash/f17d3/folder.png 1x"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
            width={250}
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/hash/f17d3/folder.png 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/hash/f17d3/folder.png 1x\\" src=\\"/static/hash/f17d3/folder.png\\" alt=\\"Another Level Two\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <div
        className="folder-title"
      >
        Another Level Two
      </div>
    </a>
  </article>,
]
`;
